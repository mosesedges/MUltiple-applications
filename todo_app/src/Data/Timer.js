import React,{useState} from 'react'

const Timer = () => {
const [count, setCount] = useState(0)

const handleClick = () => {

  setTimeout(() => {
    setCount(prev => prev + 1)
  }, 5000);
  
}

  return (
    <div>
      <h1 data-testid= 'count'>{count}</h1>
      <button data-testid='btnUp' onClick={() => handleClick()}>Increment</button>
      <button data-testid='btnDown' onClick={() => setCount(count - 1)}>Decrement</button>
      <button data-testid='btnReset' onClick={() => setCount(0)}>Reset</button>
    </div>
  )
}

export default Timer
