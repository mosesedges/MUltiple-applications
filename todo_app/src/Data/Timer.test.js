import React from 'react'
import ReactDOM from 'react-dom'
import {render, cleanup, fireEvent, wait, waitFor} from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect";
import Timer from './Timer'
import { get } from 'react-hook-form';



afterEach(cleanup);

it('should take snapshot', () => {
  const {asFragment} = render(<Timer/>)
  expect(asFragment(<Timer/>)).toMatchSnapshot()
})


it('should equal 0', () => {
  const{getByTestId} = render(<Timer/>)
  expect(getByTestId('count')).toHaveTextContent(0)
})

it('should be enabled', () => {
  const{getByTestId} = render(<Timer/>)
  expect(getByTestId('btnUp')).toBeEnabled()
})

it('should be enable ', () => {
  const{getByTestId} = render(<Timer/>)
expect(getByTestId('btnDown')).toBeEnabled()
})


it('increment count', () => {
  const{getByTestId} = render(<Timer/>)

  fireEvent.click(getByTestId('btnUp'))

  expect(getByTestId('count')).toHaveTextContent('1')
})


it('should decrement ', () => {
  const {getByTestId} = render(<Timer/>)

  fireEvent.click(getByTestId('btnDown'))
  
  expect(getByTestId('count')).toHaveTextContent('-1')
})

it('should reset', () => {
  const {getByTestId} = render(<Timer/>)

  fireEvent.click(getByTestId('btnReset'))

  expect(getByTestId('count')).toHaveTextContent(0)
})


it('increment count after 0.5s', async () => {
  const {findByTestId, getByText} = render(<Timer/>)

  fireEvent.click (awaitscreen.findByTestId('btnUp'))

  const count = await waitFor(() =>  getByText('1'))

expect(getByText(count)).toBeInTheDocument()
 

    

})