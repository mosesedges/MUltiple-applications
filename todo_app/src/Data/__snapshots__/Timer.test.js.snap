// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1
      data-testid="count"
    >
      0
    </h1>
    <button
      data-testid="btnUp"
    >
      Increment
    </button>
    <button
      data-testid="btnDown"
    >
      Decrement
    </button>
    <button
      data-testid="btnReset"
    >
      Reset
    </button>
  </div>
</DocumentFragment>
`;
